src/extractor/guidance/intersection_handler.cpp

```diff
@@ -51,6 +51,11 @@ TurnType::Enum IntersectionHandler::findBasicTurnType(const EdgeID via_edge,
     const auto &in_data = node_based_graph.GetEdgeData(via_edge);
     const auto &out_data = node_based_graph.GetEdgeData(road.eid);
 
+    // suppress instructions on edges that the can't control anyway
+    if (!in_data.startpoint && !out_data.startpoint)
```

Comment:
In my view, we shouldn't abuse is startpoint here (and below). We want to filter out `Ferry` turns. As already stated in the discussion, we should use the TravelMode Ferry to do so. Someone might decide that starting on other roads is invalid, but we know that jumping ship is not possible. If someone disables motorways as start points, we might still want to give instructions.